tools/base/fpgad/config_file.c

```diff
@@ -30,7 +30,216 @@
 
 #include "config_file.h"
 
+#include <json-c/json.h>
 
+#ifdef LOG
+#undef LOG
+#endif
+#define LOG(format, ...) \
+log_printf("cfg: " format, ##__VA_ARGS__)
 
+#define CFG_TRY_FILE(__f) \
+do { \
+	canon = canonicalize_file_name(__f); \
+	if (canon) { \
+		err = strncpy_s(c->cfgfile, \
```

Comment:
would it make sense to parameterize c->cfgfile in the macro?

Reply:
I'm not sure in this case.  Ideally we would also parameterize canon and err, etc.  I'm not sure it buys anything since the macro will only be used here.